docs/GETTING-STARTED.md

```diff
@@ -0,0 +1,424 @@
+# Getting started with Network Service Mesh
+
+This document illustrates the procedure to install and start network service mesh from a bare bones system.`
+
+## Table of Contents
+1. [Introduction](#"Introduction")
+2. [Set up Virtualization](#"Nested")
+3. [Install on Ubuntu](#"Ubuntu")
+4. [Install on Fedora.](#"Fedora")
+5. [Install pre-requisites](#"prereq")
+6. [Install docker](#"docker")
+7. [Install docker CE](#"dockerCE")
+8. [Install kubernetes](#kube)
+9. [Install virt and kvm](#virt)
+10. [Install kubectl](#kubectl)
+11. [Install golang](#golang)
+12. [Install minikube ](#minikube)
+13. [Install kvm2 driver](#kvm2)
+14. [Install protobuf](#proto)
+15. [Install NSM](#NSM)
+16. [Verify NSM](#NSMverify)
+17. [Manual steps](#Manual)
+18. [TODO](#TODO)
+19. [Credits](#credit)
+
+## Introduction
+
+For a working example we will launch a VM running kubernetes, and all required packages as well as network service mesh itself. The exam;e starts with launching a VM under Qemu/KVM. You may of course choosse a different hypervisor but those procedures haven't been verified in this document.
+
+You may use your favorite distro. In this document we hope to show the procedures for Ubuntu, Centos, Fedora, RHEL, and perhaps others.
+
+The steps here have analogies for other distros.  Also, I used minicube cluster which provides a more interesting case and probably is a closer simulation of what may happen on a "real"
+ scaled cloud, private or public.
+
+## Set up Nested Virtualization <a name="Nested"></a>
+
+For an example, install in a x86_64 VM running under Qemu/KVM.
+
+ If you choose to install on a VM, be advised that we are using minikube. minikube creates its own VM to run the cluster. You must make sure you are set up for nested virtualization.
```

Comment:
Not exactly minikube doesn't always create its own VM, there is a flag called `--vm-driver=none` which runs all components in containers. Nested virtualization is not a requirement. Thats how we run in Travis today.

Reply:
Yes, in the getting started guide, I documented running minikube with the kvm driver as I thought that might give independence from the environment on the current system which may have other clusters running. I will add some lines to document it both ways.